pkg/repo/repotest/server.go

```diff
@@ -185,31 +169,39 @@ func (srv *OCIServer) Run(t *testing.T, opts ...OCIServerOpt) {
 		t.Fatal("error removing chart before push")
 	}
 
-	err = registryClient.SaveChart(ch, ref)
+	// save it back to disk..
+	absPath, err := chartutil.Save(ch, srv.Dir)
+	if err != nil {
+		t.Fatal("could not create chart archive")
+	}
+
+	// load it into memory...
+	contentBytes, err := ioutil.ReadFile(absPath)
 	if err != nil {
-		t.Fatal("error saving chart")
+		t.Fatal("could not load chart into memory")
 	}
 
-	err = registryClient.PushChart(ref)
+	_, err = registryClient.Push(contentBytes, ref)
 	if err != nil {
-		t.Fatal("error pushing chart")
+		t.Fatalf("error pushing dependent chart: %s", err)
 	}
 
 	if cfg.DependingChart != nil {
```

Comment:
the nested `if`s has me wondering if it is possible/reasonable to reverse the logic and return if `DependingChart` is nil.  it would likely require the assignment L208 to come before this check / return.  Not a bigger... this is readable.

Reply:
Resolved in 30a8ce59861a4d9fc935d8e1dd53664208876621